app/models/calnet_ldap/person.rb

```diff
@@ -0,0 +1,75 @@
+module CalnetLdap
+  # https://calnetweb.berkeley.edu/calnet-technologists/ldap-directory-service/how-ldap-organized/people-ou
+  class Person
+    def self.get(user)
+      if net_ldap_entry = CalnetLdap::Client.new.search_by_uid(user.uid)
+        return self.new(net_ldap_entry)
+      end
+      nil
+    end
+
+    def initialize(net_ldap_entry)
+      if net_ldap_entry.respond_to?(:attribute_names)
+        net_ldap_entry.attribute_names.each do |key|
+          instance_var_id = "@#{key.to_s}"
+          instance_variable_set(instance_var_id, net_ldap_entry[key]) if !instance_variable_defined?(instance_var_id)
+        end
+      end
```

Comment:
Nice bit of metaprogramming, but is there a reason we couldn't just use some private accessors for the incoming attributes?